index.js

```diff
@@ -62,15 +61,28 @@ const useNavigate = (options) => {
  * Part 2, Low Carb Router API: Router, Route, Link, Switch
  */
 
-export const Router = (props) => {
-  // this little trick allows to avoid having unnecessary
-  // calls to potentially expensive `buildRouter` method.
+export const Router = ({ hook, matcher, base = "", nested = false, children }) => {
+  const parent = useRouter();
+
+  // nested `<Router />` has the scope of its closest parent router (base path is prepended)
+  // Routers are not nested by default, but this might change in future versions
+  const proto = nested ? parent : defaultRouter;
+
+  // updates the current router with the props passed down to the component
+  const updateRouter = (router) => {
+    router.hook = hook || proto.hook;
+    router.matcher = matcher || proto.matcher;
+    router.base = proto.base + base;
+  };
+
+  // we use `useState` here, but it only catches the first render and never changes.
   // https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily
-  const [value] = useState(() => ({ v: buildRouter(props) }));
+  const [value] = useState(() => ({})); // create the object once...
+  updateRouter(value); // then update it on each render
```

Comment:
@molefrog this will work on React 17, but React 18 uses concurrent rendering so we really need this inside a `useLayoutEffect()` to guarantee the correct ordering. Otherwise, these properties may not be updated to reflect the current render, as renders may occur "out of order", or be cancelled. I mentioned this also in my PR in response to a (now-resolved) comment asking the same question.